### DISCUSSION TITLE
Can modify libvips memory allocation

### DISCUSSION THREAD

**Discussion by zhaohuxing** [libvips/libvips#2931] [Help wanted]
Hi @jcupitt 

I use `libvips-binding-rust` memory allocation by jemalloc, `libvips-binding-rust` use memory allocation by `gslice` and `glibc(malloc)`  etc。my application crashed, print error log
```
process:6752): GLib-GObject-WARNING **: 15:57:04.025: invalid uninstantiatable type '(null)' in cast to 'VipsObject'
(process:6752): GLib-GObject-CRITICAL **: 15:57:04.025: g_object_ref: assertion 'G_IS_OBJECT (object)' failed
(process:6752): GLib-GObject-WARNING **: 15:57:04.025: invalid uninstantiatable type '(null)' in cast to 'VipsObject'
(process:13376): GLib-GObject-CRITICAL **: 16:07:05.654: g_closure_unref: assertion 'closure->ref_count > 0' failed
(process:13376): GLib-GObject-CRITICAL **: 16:07:05.654: g_closure_unref: assertion 'closure->ref_count > 0' failed
(process:13376): GLib-GObject-CRITICAL **: 16:07:05.654: g_closure_ref: assertion 'closure->ref_count > 0' failed
(process:13376): GLib-GObject-CRITICAL **: 16:07:05.654: g_closure_invoke: assertion 'closure->marshal || real_closure->meta_marshal' failed
(process:13376): GLib-GObject-CRITICAL **: 16:07:05.654: g_closure_unref: assertion 'closure->ref_count > 0' failed
***MEMORY-ERROR***: [11237]: GSlice: assertion failed: aligned_memory == (gpointer) addr
```
so, I guess it is caused by inconsistent use of memory allocation,
so, how to modify the libvips memory allocation method?

**Comment by jcupitt** [comment]
Hello again,

I think you are probably seeing a refcount error in the rust bindings. I would open an issue on that project.

**Reply by zhaohuxing** [reply]
this is https://github.com/olxgroup-oss/libvips-rust-bindings 
😊

**Reply by jcupitt** [reply]
Yes, I think they have a bug. I would open an issue on that project.

**Reply by zhaohuxing** [reply]
@jcupitt 
I didn't find your issue, could you give me the link?

**Reply by jcupitt** [reply]
I think you asked in the right place: https://github.com/olxgroup-oss/libvips-rust-bindings/issues/26

**Reply by zhaohuxing** [reply]
Thanks @jcupitt
